### PR TITLE
fix: debounce the item selector query in dashboard editor [DHIS2-10530]

### PR DESCRIPTION
**Pull request by jenniferarnesen**
Fix:
- prevent the item selector query from firing on every keystroke.

Refactor:
- useDataEngine instead of manually constructing the query
- SinglesMenuGroup exports was triggering eslint warning, so no longer export const SinglesMenuGroup (unit test had to be updated)
- remove USERS from ItemSelector as it is no longer meant to be used (confirmed with @janhenrikoverland )
- borrow the `useDebounce` hook from analytics. Have suggested to Austin that it could be added to app-runtime

### CODE REVIEW

File: src/components/ItemSelector/ItemSelector.js [hunk 1]
```diff
@@ -14,21 +16,17 @@ const ItemSelector = () => {
     const [filter, setFilter] = useState('')
     const [items, setItems] = useState(null)
     const [maxOptions, setMaxOptions] = useState(new Set())
-    const { d2 } = useD2({})
+    const dataEngine = useDataEngine()
+    const debouncedFilterText = useDebounce(filter, 200)
```
Comment (janhenrikoverland) [inline src/components/ItemSelector/ItemSelector.js]: I think we should set 500 to get more out of it @jenniferarnesen @cooper-joe

**Review by janhenrikoverland — APPROVED** [review]
Approved when comment is resolved.

File: src/components/ItemSelector/ItemSelector.js (hunk 1 above)
Comment (cooper-joe) [reply src/components/ItemSelector/ItemSelector.js]: I would argue that 500ms is too long. For form validation the debounce can be longer, but for search the user is expecting to see results, so any delay is purely technical in nature. I would say 350ms should be the maximum, unless there's a strong technical backing to delay it further. (Related: [Doherty Threshold](https://lawsofux.com/doherty-threshold/))

Comment (janhenrikoverland) [reply src/components/ItemSelector/ItemSelector.js]: Good point, @cooper-joe. Let's go with 350 ms.

**Pull request merged**